server-impl.js

```diff
@@ -1,166 +1,218 @@
-
-    var authorization = req.header('Authorization');
+app.get("/:key/securequery/:subject/:relationship", function (req, res) {
+  var authorization = req.header("Authorization");
 
-    if ((typeof authorization == 'undefined')) {
-        res.status(401).send({'status': 'ERROR', 'message': 'Missing Authorization header!'});
-    } else {
-        var data = {
-            'query': req.query,
-            'body': {}
-        };
-
-        processQuery(req.params.key, req.params.subject, req.params.relationship, data, req, res);
-    }
-});
-
-app.post('/:key/securequery/:subject/:relationship', function(req, res) {
-
-    var authorization = req.header('Authorization');
-
-    if ((typeof authorization == 'undefined')) {
-        res.status(401).send({'status': 'ERROR', 'message': 'Missing Authorization header!'});
-    } else {
-        var data = {
-            'query': req.query,
-            'body': req.body
-        };
+  if (typeof authorization == "undefined") {
+    res
+      .status(401)
+      .send({ status: "ERROR", message: "Missing Authorization header!" });
+  } else {
+    var data = {
+      query: req.query,
+      body: {},
+    };
 
-        processQuery(req.params.key, req.params.subject, req.params.relationship, data, req, res);
-    }
+    processQuery(
+      req.params.key,
+      req.params.subject,
+      req.params.relationship,
+      data,
+      req,
+      res
+    );
+  }
 });
 
-app.post('/:key/query/:subject/:relationship', function(req, res) {
+app.post("/:key/securequery/:subject/:relationship", function (req, res) {
+  var authorization = req.header("Authorization");
 
+  if (typeof authorization == "undefined") {
+    res
+      .status(401)
+      .send({ status: "ERROR", message: "Missing Authorization header!" });
+  } else {
     var data = {
-        'query': req.query,
-        'body': req.body
+      query: req.query,
+      body: req.body,
     };
 
-    processQuery(req.params.key, req.params.subject, req.params.relationship, data, req, res);
+    processQuery(
+      req.params.key,
+      req.params.subject,
+      req.params.relationship,
+      data,
+      req,
+      res
+    );
+  }
 });
 
-app.post('/rbproxy',  require('./handlers/rbproxy').process);
+app.post("/:key/query/:subject/:relationship", function (req, res) {
+  var data = {
+    query: req.query,
+    body: req.body,
+  };
+
+  processQuery(
+    req.params.key,
+    req.params.subject,
+    req.params.relationship,
+    data,
+    req,
+    res
+  );
+});
 
+app.get("/mock-response/:code", function (req, res) {
+  const code = req.params.code;
+  const data = { conditions: "sunny", temperature: 30 };
+  const regex = /2[0-9][0-9]/g;
+
+  if (regex.test(code)) {
+    res.status(code);
+    res.setHeader("Content-Type", "application/json");
+    res.send(JSON.stringify({ status: "OK", data }, null, 3));
+  } else res.status(401).send({ status: "ERROR", message: "Unhandled Status Code" });
```

Comment:
Not a biggie David as it's a test app but a 400 status may have been more suitable here than a 401.

Reply:
Thanks David :-)